src/main/java/de/zalando/aruha/nakadi/service/subscription/zk/CuratorZkSubscriptionClient.java

```diff
@@ -134,19 +137,20 @@ public void fillEmptySubscription(final Map<Partition.PartitionKey, Long> partit
         }
     }
 
-    private static byte[] serializeNode(final String session, final String nextSession, final Partition.State state) {
+    private static byte[] serializeNode(@Nullable final String session, @Nullable final String nextSession,
+                                        final Partition.State state) {
         return Stream.of(
-                session == null ? "" : session,
-                nextSession == null ? "" : nextSession,
+                session == null ? NO_SESSION : session,
+                nextSession == null ? NO_SESSION : nextSession,
                 state.name()).collect(Collectors.joining(":")).getBytes(CHARSET);
     }
 
     private static Partition deserializeNode(final Partition.PartitionKey key, final byte[] data) {
         final String[] parts = new String(data, CHARSET).split(":");
         return new Partition(
                 key,
-                parts[0].isEmpty() ? null : parts[0],
-                parts[1].isEmpty() ? null : parts[1],
+                NO_SESSION.equals(parts[0]) ? null : parts[0],
```

Comment:
@adyach I saw your question in email, but for some reason can't find it here. 
`parts[0]` and 'parts[1]' will never be null, as we don't write nulls there, only real values or empty string. So we can expect the same when we read data.